File: PrezBDDTDD.Tests.JavaScript/Process/ServiceVirement.js
function ServiceVirement() {
    //Todo : demander la limite de plafond par défaut
    this.Plafond = 1000;
};

ServiceVirement.prototype = {
    EffectuerVirement: function(/*int*/ montant, /*Compte*/ compteDebiteur, /*Compte*/ compteCrediteur) {
        if (montant > compteDebiteur.Solde)
        {
            return RetourVirement.SoldeDepasse;
        }

        if (montant > this.Plafond)
        {
            return RetourVirement.PlafondDepasse;
        }

        compteDebiteur.Solde -= montant;
        compteCrediteur.Solde += montant;
        return RetourVirement.Ok;
    }
};

File: PreszBDDTDD.Tests.Java/src/main/java/presz/bddtdd/tests/java/process/ServiceVirement.java
package presz.bddtdd.tests.java.process;

import presz.bddtdd.tests.java.model.Compte;

public class ServiceVirement {

    private int plafond;

    public ServiceVirement() {
        //Todo : demander la limite de plafond par défaut
        this.plafond = 1000;
    }

    public RetourVirement effectuerVirement(int montant, Compte compteDebiteur, Compte compteCrediteur) {
        if (montant > compteDebiteur.getSolde()) {
            return RetourVirement.SoldeDepasse;
        }
        if (montant > plafond) {
            return RetourVirement.PlafondDepasse;
        }

        compteDebiteur.setSolde(compteDebiteur.getSolde() - montant);
        compteCrediteur.setSolde(compteCrediteur.getSolde() + montant);
        return RetourVirement.Ok;
    }

    public int getPlafond() {
        return plafond;
    }

    public void setPlafond(int plafond) {
        this.plafond = plafond;
    }
}
